src/agent/rest_web_server.cpp

```diff
@@ -0,0 +1,1061 @@
+#include <fstream> 
+#include <string>
+#include <sstream>
+#include <errno.h>
+#include <sys/time.h>
+#include <string>
+#include "openthread/thread_ftd.h"
+#include "common/logging.hpp"
+
+#include "agent/rest_web_server.hpp"
+
+#include "cJSON.h"
+#include "cJSON.c"
+#include "http_parser.h"
+#include "http_parser.c"
+
+
+
+namespace otbr {
+namespace agent {
+
+
+static int on_message_begin(http_parser *parser) {
+    Connection *Connection = (struct Connection*)parser->data;
+    
+    otbrLog(OTBR_LOG_ERR, "complete: %d",Connection->complete );
+    return 0;
+}
+
+static int on_status(http_parser *parser,const char *at, size_t len) {
+    Connection *Connection = (struct Connection*)parser->data;
+    Connection->status = new char[len+1]();
+    strncpy(Connection->status,at,len);
+    return 0;
+}
+
+
+static int on_url(http_parser *parser, const char *at, size_t len) {
+    Connection *Connection = (struct Connection*)parser->data;
+    Connection->path = new char[len+1]();
+    strncpy(Connection->path,at,len);
+    return 0;
+}
+
+static int on_header_field(http_parser *parser, const char *at, size_t len) {
+    Connection *Connection = (struct Connection*)parser->data;
+    Connection->header_f = new char[len+1]();
+   
+    
+    strncpy(Connection->header_f,at,len);
+    otbrLog(OTBR_LOG_ERR, "header_f: %s",Connection->header_f);
+    return 0;
+}
+
+static int on_header_value(http_parser *parser, const char *at, size_t len) {
+    Connection *connection = (struct Connection*)parser->data;
+    connection->header_v = new char[len+1]();
+    
+    strncpy(connection->header_v,at,len);
+    otbrLog(OTBR_LOG_ERR, "header_v: %s",connection->header_v);
+    return 0;
+}
+static int on_body(http_parser *parser, const char *at, size_t len) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->body = new char[len+1]();
+    strncpy(connection->body,at,len-1);
+    
+    otbrLog(OTBR_LOG_ERR, "body : %s",connection->body);
+    return 0;
+}
+
+static int on_headers_complete(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->content_length = parser->content_length;
+    connection->method = parser->method;
+    return 0;
+}
+
+static int on_message_complete(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->complete = 0;
+    return 0;
+}
+
+static int on_chunk_header(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->complete = 0;
+    return 0;
+}
+
+static int on_chunk_complete(http_parser *parser) {
+    Connection *connection =(struct Connection*) parser->data;
+    connection->complete = 0;
+    return 0;
+}
+
+static int setNonBlocking(int fd)
+{
+	int oldflags;
+
+	oldflags = fcntl(fd, F_GETFL);
+
+	if (fcntl(fd, F_SETFL, oldflags | O_NONBLOCK) == -1) {
+		return -1;
+	}
+
+	return 0;
+}
+
+static http_parser_settings mSettings {
```

Comment:
We use `m` monation for only member fields of class/struct. 
```suggestion
static http_parser_settings sSettings {
```

`s` stands for `static`.